Mathlib/GroupTheory/FreeGroup/CyclicallyReduced.lean

```diff
@@ -73,6 +74,25 @@ theorem flatten_replicate (n : ℕ) (h : IsCyclicallyReduced L) :
 
 end IsCyclicallyReduced
 
+@[to_additive]
+theorem IsReduced.append_flatten_replicate_append (n : ℕ) (hn : n ≠ 0) (L₁ L₂ L₃ : List (α × Bool))
```

Comment:
I might suggest putting `n` and `hn` at the end of the variables? Also, have you given any thought to whether any of these variables should be implicit?